src/compiler/checker.ts

```diff
@@ -4531,7 +4531,7 @@ namespace ts {
             return links.resolvedType;
         }
 
-        function getStringLiteralType(node: StringLiteral): StringLiteralType {
+        function getStringLiteralType(node: StringLiteral | StringLiteralTypeNode): StringLiteralType {

```

Comment:
Maybe this should just take the `string` and call it `getStringLiteralTypeForText`; what are your thoughts?